modules/client-handler/src/main/java/org/apache/ignite/client/handler/JdbcQueryEventHandlerImpl.java

```diff
@@ -398,6 +481,10 @@ void close() {
         }
 
         <T> CompletableFuture<T> doInSession(SessionAwareAction<T> action) {
+            if (closed) {
```

Comment:
I don't see how this helps against concurrent connection closing

Reply:
Thank you. Slightly reworked, currently I just prevent the session from being re-created after the context is closed.